subprojects/kotlin-dsl-provider-plugins/src/main/kotlin/org/gradle/kotlin/dsl/provider/plugins/precompiled/tasks/CompilePrecompiledScriptPluginPlugins.kt

```diff
@@ -33,10 +32,10 @@ import org.gradle.kotlin.dsl.support.compileKotlinScriptModuleTo
 
 
 @CacheableTask
-open class CompilePrecompiledScriptPluginPlugins : ClassPathSensitiveTask() {
+abstract class CompilePrecompiledScriptPluginPlugins : ClassPathSensitiveTask(), SharedAccessorsPackageAware {
 
     @get:OutputDirectory
-    var outputDir = directoryProperty()
+    val outputDir = directoryProperty()
```

Comment:
:o: Prefer declarative abstract task property

This should be done on the set of tasks changed in this PR. Not blocking merging this PR.

```suggestion
    abstract val outputDir: DirectoryProperty
```